Expense/expenseRouter.js

```diff
@@ -0,0 +1,32 @@
+const express = require('express');
+
+const Expense = require('./Expense.js');
+
+const router = express.Router();
+
+router.get('/', (req, res) => {
+    Expense.find()
+    .select("description amount -_id")
+    .populate('budget', "title budgetAmount -_id")
+    .populate('category', "title -_id")
+    .then((expense
+    ) => {
+        res.status(200).json(expense
+        )
+    })
+})
+
+router.post('/', (req, res) => {
```

Comment:
Instead of sending the budget and category `ObjectId` along with the `POST` request, we can send a `POST` request that looks like this:

```
{
	"amount": 35,
	"description": "potatoes",
	"budget": "Monthly Spending",
	"category": "Groceries"
}
```

In order to do that, we need to add some logic to our route:

```
.post((req, res, next) => {
  let { amount, budget, category, description } = req.body

  budget = Budget.findOne({ title: budget })
  category = Category.findOne({ title: category })

  Promise.all([budget, category])
    .then(([budget_id, category_id]) => {
      const expense = new Expense({
        amount,
        budget: budget_id,
        category: category_id,
        description
      })

      expense.save()
        .then(response => res.status(201).json(response))
        .catch(err => next(err))
    })
    .catch(err => next(err))
  })
```

We have to get `budget` and `category` with the `findOne` method, then using `Promise.all`, we grab the `ObjectId` for each from the response from MongoDB, then create our new Expense.